packages/tools/src/tools/displayTools/Labelmap/addLabelmapToElement.ts

```diff
@@ -61,10 +61,13 @@ async function addLabelmapToElement(
   } else {
     // We can use the current imageId in the viewport to get the segmentation imageId
     // which later is used to create the actor and mapper.
-    const segmentationImageId = (
+    const segmentationImageIds = (
       labelMapData as LabelmapSegmentationDataStack
     ).imageIdReferenceMap.get(viewport.getCurrentImageId());
 
+    // Todo: fix this to support multiple derived imageIds
```

Comment:
Can you revert the change and instead allow multiple segmentation UIDs to be associated?  That way they are cleanly separated.  With this change it becomes very hard to know where to do things.